backend-rails/spec/models/service_order_spec.rb

```diff
@@ -31,4 +31,86 @@
   context 'inclusion' do
     it { should validate_inclusion_of(:status).in_array(ServiceOrder::STATUSES) }
   end
+
+  describe '#products' do
+    let(:service_order) { create(:service_order) }
+    let(:products) { service_order.products }
+
+    context 'when service order does not have a product' do
+      it 'returns an empty array' do
+        expect(products).to eq([])
+      end
+    end
+
+    context 'when service order has a product' do
```

Comment:
```suggestion
    context 'when service order has products' do
```

If I understood you only created 2 products, so this means that if we replace the product method for `Item.all` this spec will pass, is this correct?

Reply:
I don't think so, because the method `product` is selecting only items that have `item_type = product`. This method has product and service, so `Item.all` will return all types and not just product type. However, I'll check this.